staging/src/k8s.io/apiserver/pkg/apis/config/types.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LeaderElectionConfiguration defines the configuration of leader election
+// clients for components that can run with leader election enabled.
```

Comment:
does this belong in apiserver? lots of things that aren't apiservers use leader election. I expected it in apimachinery

Reply:
As it discussed it in  [doc](https://docs.google.com/document/d/1-u2y03ufX7FzBDWv9dVI_HyiIQZL8iz3o3vabeOpP5Y/edit#), in apiserver part, LeaderElectionConfiguration was in it. 
@luxas @sttts can you show some info?

Reply:
We were undecided about this one. It seems to be higher level than apimachinery, but also has no tight connections to apiserver. I don't feel strongly with either direction. As the types are kube independent, we can technically also move it to apimachinery. I would like to see some generic `ResourceLock` syntax which does not reference kube.

Reply:
cc @mtaufen @wojtek-t who were working on the Lease API

Reply:
> does this belong in apiserver? lots of things that aren't apiservers use leader election. I expected it in apimachinery

I don't see this as related to our type system in the same way as `TypeMeta`, `ObjectMeta`, `Time`, or `List`.  I don't even see it as quite as fundamental as `ListOptions`.  Something common for many bits of config, sure.  Something common for all types/schemes, not really.

Reply:
If I understand correctly, components can migrate to using the Lease API (it was designed to represent the same information as LeaderElectionRecord), but they still have to get related configuration from somewhere (i.e. LeaderElectionConfiguration).

So this is a question of where to put a common config struct type?

Reply:
is this resolved?

Reply:
Yes, I think the consensus is to put it here, in k8s.io/apiserver vs putting it in k8s.io/apimachinery which should hold even more generic types.